src/ucp/wireup/wireup.c

```diff
@@ -738,7 +738,13 @@ void ucp_wireup_assign_lane(ucp_ep_h ep, ucp_lane_index_t lane, uct_ep_h uct_ep,
         ucs_trace("ep %p: wireup uct_ep[%d]=%p next set to %p%s", ep, lane,
                   ep->uct_eps[lane], uct_ep, info);
         ucp_wireup_ep_set_next_ep(ep->uct_eps[lane], uct_ep);
-        ucp_wireup_ep_remote_connected(ep->uct_eps[lane]);
+        /* Don't mark a WIREUP EP as remote connected in case of CM, since it
+         * flushes and removes a WIREUP EP, but a WIREUP EP is used to keep
+         * user's pending requests and send WIREUP MSGs until CM and WIREUP_MSG
+         * phases are not done */
+        if (!ucp_ep_has_cm_lane(ep)) {
```

Comment:
maybe pass "remote_connected" boolean parameter from ucp_wireup_connect_lane_to_iface to ucp_wireup_assign_lane?
in case of ucp_wireup_connect_lane_to_iface() it will be 0, in other cases - 1

Reply:
`ucp_wireup_assign_lane()` is used only in `ucp_wireup_connect_lane_to_iface()`

Reply:
never set remote connected, replace by assert(has_cm)

Reply:
remove this function ucp_wireup_assign_lane altogether, merge into ucp_wireup_connect_lane_to_iface

Reply:
fixed